apps/design/frontend/src/geography_screen.test.tsx

```diff
@@ -308,26 +316,25 @@ describe('Precincts tab', () => {
       );
     }
 
-    const electionWithNewPrecinctRecord: ElectionRecord = {
-      ...electionWithNoPrecinctsRecord,
-      election, // Don't update the election, since it's not used here and it's hard to update ballot styles correctly
-      precincts: [newPrecinct],
-    };
-    apiMock.updatePrecincts
+    apiMock.createPrecinct
       .expectCallWith({
         electionId,
-        precincts: [newPrecinct],
+        newPrecinct,
       })
       .resolves();
-    apiMock.getElection
-      .expectCallWith({ user, electionId })
-      .resolves(electionWithNewPrecinctRecord);
+    apiMock.listPrecincts
+      .expectCallWith({ electionId })
+      .resolves([newPrecinct]);
+    // Districts haven't changed, but we are using coarse-grained invalidation
```

Comment:
Is it necessary to use coarse invalidation ie. `invalidateElectionQueries` when only precincts have changed? Changing districts may change precincts, but not vice versa.

OTOH I see the value of simplifying because precincts and districts both are stored in the `elections` table, cache invalidation is hard, and the performance change (if any) is probably not noticeable.

Reply:
I was thinking along the lines of your second paragraph - it's hard to get cache invalidation right, so it makes sense to start with the safest thing and optimize as needed.